index/encoding/doc/doc.go

```diff
@@ -80,8 +80,9 @@ func (w *writer) Open() error {
 }
 
 func (w *writer) Write(d doc.Document) error {
-	w.enc.PutUvarint(uint64(len(d.Fields)))
+	w.enc.PutBytes(d.ID)
```

Comment:
The encoder/decoder is beginning to look super similar to the types in https://github.com/m3db/m3db/blob/master/serialize/types.go 

Any chance we can consolidate?

Reply:
I'm not opposed to consolidating though given that they live in separate repos that may be difficult at the moment. Perhaps we can track in an issue?

Reply:
Sure, https://github.com/m3db/m3ninx/issues/36